### PR TITLE
v4 feat(logging): New logger

### PR DESCRIPTION
**Pull request by tobiastimm**
Done:

- [x] Added colors & styles for console output
- [x] Combine logger and styling
- [x] Spinner

Things to do:

- [ ] More tests

### CODE REVIEW

File: src/core/interfaces.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+
 export interface IConfig {
   root?: string;
   target?: 'browser' | 'server' | 'electron' | 'universal';
   homeDir?: string;
   output?: string;
   modules?: Array<string>;
   fuseBoxPolyfillsFolder?: string;
-
+  logging?: ILoggerProps;
```
Comment (nchanged) [inline src/core/interfaces.ts]: Do you think we could add `boolean` there? So user could just disable all the logs ?

**Comment by tobiastimm** [comment]
Should we merge this and add more tests later? So that it is already useable? @nchanged

**Comment by nchanged** [comment]
@tobiastimm can you rebase please? ;-) let's merge it!

**Comment by nchanged** [comment]
Context is still removed from your branch ;-(

**Comment by tobiastimm** [comment]
yeah sorry, rebasing isn't my thing. I hate it

**Comment by nchanged** [comment]
Good job! merging it ;-)

File: src/core/context.ts~feat(logging): Integrate with context [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { createConfig } from './config';
```
Comment (nchanged) [inline src/core/context.ts~feat(logging): Integrate with context]: what is this file? xDD

**Pull request merged**